include/vast/Translation/CodeGenUnsupportedVisitor.hpp

```diff
@@ -0,0 +1,527 @@
+// Copyright (c) 2023-present, Trail of Bits, Inc.
+
+#pragma once
+
+#include "vast/Util/Warnings.hpp"
+#include "vast/Translation/CodeGen.hpp"
+#include "vast/Translation/CodeGenContext.hpp"
+#include "vast/Translation/CodeGenVisitor.hpp"
+#include "vast/Translation/CodeGenBuilder.hpp"
+#include "vast/Translation/CodeGenFallBackVisitor.hpp"
+
+namespace vast::cg
+{
+
+#define VISITOR_DECL(decl_type) \
+	operation Visit##decl_type(const clang::decl_type *decl) { \
+		return make_decl_empty< hl::UnsupportedDecl >(decl); \
+	}
+
+#define VISITOR_STMT(type) \
+	operation Visit##type(const clang::type *expr) { \
+		return make_default_operation< vast::hl::UnsupportedExpr >(expr); \
+	}
+
+	template< typename Derived >
+	struct UnsupportedDeclVisitor
+		: clang::ConstDeclVisitor< UnsupportedDeclVisitor< Derived >, vast::Operation* >
+		, vast::cg::CodeGenVisitorLens< UnsupportedDeclVisitor< Derived >, Derived >
+		, vast::cg::CodeGenBuilderMixin< UnsupportedDeclVisitor< Derived >, Derived >
+	{
+		using LensType = vast::cg::CodeGenVisitorLens< UnsupportedDeclVisitor< Derived >, Derived >;
+
+		using LensType::derived;
+		using LensType::context;
+		using LensType::mcontext;
+		using LensType::acontext;
+
+		using LensType::meta_location;
+		using LensType::visit;
+
+		template< typename Op, typename... Args >
```

Comment:
I am pretty sure all of these helpers are copied from other parts of `cg::` (or at least they look similar-ish enough to be considered a good candidates for shared utilities) - cannot they be re-used?

Reply:
Most of these utility functions are custom and written similar to one in `cg::Builder`. These utility function buildup region and generate operation specific to the information we need to present with the unsupported operations. Moving them to the common place in builder will not be a good idea. 

I may refactor them a few and see if they can use already existing utility methods from `cg::Builder`.